plugins/nvme/fu-nvme-device.c

```diff
@@ -109,7 +115,26 @@ fu_nvme_device_submit_admin_passthru (FuNvmeDevice *self, struct nvme_admin_cmd
 			     strerror (errno));
 		return FALSE;
 	}
-	return TRUE;
+
+	/* check the error code */
+	err = rc & 0x3ff;
```

Comment:
Is this logic correct?  Shouldn't you be looking at `cmd->result` for this information if `rc>=0`?

Reply:
I agree it's wonky, but it's what the kernel provides, and matches what nvme-cli does.